helion/runtime/kernel.py

```diff
@@ -291,6 +291,7 @@ def __init__(self, kernel: Kernel[_R], args: tuple[object, ...]) -> None:
         """
         super().__init__()
         self.kernel = kernel
+        self.args = args
```

Comment:
Won't this cause a memory leak?  Now the first args to the kernel will never get deallocated.  (This is why it was done the prior way.)